File: Code/Erlang/problem001/euler001.erl
-module(euler001).
-export([euler1/0]).

%% 1> c(euler001).
%% {ok,euler001}
%% 2> euler001:euler1().
%% 234168

euler1() ->
  lists:foldl( fun(X, Sum) -> X + Sum end, 0, 
    [X || X <- lists:seq(3,1000), (X rem 3 =:= 0) or (X rem 5 =:= 0)]).

File: Code/Java/problem001/Euler001.java

public class Euler001 {

  private int compute() {
    int sum = 0;
    for (int i = 2; i < 1000; ++i) {
      if ( (0 == i%3) || (0 == i%5) ) sum += i;
    }
    return sum;
  }

  public static void main(String[] args) {
    int sum = new Euler001().compute();
    System.out.println("The sum is " + sum);
  }

}
